skia/recorder/src/extractor.cpp

```diff
@@ -0,0 +1,170 @@
+#include "extractor.hpp"
+
+RiveFrameExtractor::RiveFrameExtractor(const char* path,
+                                       const char* artboard_name,
+                                       const char* animation_name,
+                                       const char* watermark_name)
+{
+	riveFile = getRiveFile(path);
+	artboard = getArtboard(artboard_name);
+	animation = getAnimation(animation_name);
+	animation_instance = new rive::LinearAnimationInstance(animation);
+	watermarkImage = getWaterMark(watermark_name);
+	rasterSurface = SkSurface::MakeRaster(SkImageInfo::Make(
+	    width(), height(), kRGBA_8888_SkColorType, kPremul_SkAlphaType));
+	rasterCanvas = rasterSurface->getCanvas();
+	ifps = 1.0 / animation->fps();
+};
+
+sk_sp<SkImage> RiveFrameExtractor::getWaterMark(const char* watermark_name)
+{
+	// Init skia surfaces to render to.
+	sk_sp<SkImage> watermarkImage;
+	if (watermark_name != NULL && watermark_name[0] != '\0')
+	{
+
+		if (!file_exists(watermark_name))
+		{
+			throw std::invalid_argument(string_format(
+			    "Cannot find file containing watermark at %s", watermark_name));
+		}
+		if (auto data = SkData::MakeFromFileName(watermark_name))
+		{
+			watermarkImage = SkImage::MakeFromEncoded(data);
+		}
+	}
+	return watermarkImage;
+}
+
+rive::File* RiveFrameExtractor::getRiveFile(const char* path)
+{
+	FILE* fp = fopen(path, "r");
+
+	if (fp == nullptr)
```

Comment:
Gotta double check the docs, but `fopen()` might need to be checked against `NULL` instead of `nullptr`